### PR TITLE
Fix notebook markup and improvements

### PR DESCRIPTION
**Pull request by santiago-imelio**
- Use correct markup on hyperlinks
- Pass only training data to the grid search
- Add final evaluation on test set

### CODE REVIEW

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:
